content/docs/user-guide/testing/parallel-pattern/_index.md

```diff
@@ -0,0 +1,476 @@
+{{ /important }}
+
+### When to enable batching or parallelization
+
+Be careful when placing a test in a batch, parallel, or shared pool! All tests should be atomic and not modify Editor state in ways which would interfere with _other_ tests that may run in the same Editor. 
+
+Due to the lack of support for setup, teardown, or test wrapping in batch, parallel, and shared tests, avoid placing any tests that rely on this support into a shared pool. 
+
+Examples of tests which can be batched, but not parallel is a physics test that makes a modification to the project wide material library at the start and restores to the default setting when finishing. By changing the state of a resource shared between Editors (disk storage), it could cause parallel tests to fail.
+
+In general, tests that require interacting with I/O may have these problems. As a general rule, it's better to make your test do changes in memory rather than on disk.
+
+### Command-line settings to configure test modes
+
+Certain test modes can be disabled or managed through command-line options:
+
+* `--no-editor-parallel`: Don't run tests in parallel. Parallel tests will become serial tests and shared tests will become batched.
+* `--no-editor-batch`: Don't batch tests, All tests run in a single editor. Shared tests will become parallel, or if parallel tests are also disabled, become serial.
+* `--editor-parallel`: Overrides the number of maximum editors run in parallel. Setting this option to `1` is equivalent to `--no-editor-parallel`.
+
+## Best practices
+
+* **Don’t create a level unless necessary**. Use an already existing empty level and don’t save the changes once the test is over.
+* **Tests should be be self-contained**. No external tools should determine if the test has passed or failed. Instead, the Editor should do this work. Use the one of the Editor return codes as result of the test, `0x0` for success and `0xF` for failure.
+* **Don’t read the Editor log file directly**. To read Editor log information else use the `DebugTraceBus` and `Tracer` utilities. The `DebugTraceBus` is able to capture any of these, without having to create an I/O dependency on a file.
+    
+    ```python
+with Tracer() as section_tracer:
+    # Do stuff
+
+# section_tracer now contains all prints,errors and warnings that happened inside the block
+if section_tracer.has_errors:
+        assert False, "AZ_Errors happened during execution"
+```
+* **Tests must be deterministic **. The application will run at different FPS and take different amounts of time time loading assets. When running the test make sure it is deterministic with these conditions.
+* **Wait on conditions instead of waiting for time.** Some tests require performing an action and waiting for completion. To optimize the amount of time your test runs, and avoiding false negatives caused by timeouts, wait on a condition instead of for an arbitrary time:
```

Comment:
```suggestion
* **Wait on conditions instead of waiting for time.** Some tests require performing an action and waiting for completion. To optimize the amount of time your test runs, and avoid false negatives caused by timeouts, wait on a condition instead of a set time:
```